### PR TITLE
Fix/2708 pasting tables

### PR DESCRIPTION
**Pull request by max-nextcloud**
### 📝 Summary

Fixes https://github.com/nextcloud/text/issues/2708.

Try to read the first table row as headings.

Pasting works via `insertContent` while opening uses `setContent`.
They use the schema in different ways.
So we also need to make sure to test both for some corner cases.

`setContent` is fairly flexible in turning the input
into a valid document structure.
`insertContent` however fails to resolve structures
that would require picking lower priority parent elements.

Note: Some tests in src/tests/nodes/Table.spec.js
fail when using `insertContent` instead of `setContent`.
Pasting the correponding html table is fixed never the less.

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation is not required

### CODE REVIEW

File: src/tests/markdown.spec.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import spec from "./fixtures/spec"
 import markdownit from './../markdownit'
 import { typesAvailable } from './../markdownit/callouts'
-import { markdownThroughEditor, markdownThroughEditorHtml } from "./helpers";
+import {
+	markdownThroughEditor,
+	markdownThroughEditorHtml,
+	markdownFromPaste
+} from "./helpers";
```
Comment (mejo-) [inline src/tests/markdown.spec.js:8]: Nitpicking, but this probably needs adjustments to please the linter (single quotes, no semicolon).

**Comment by mejo-** [comment]
Probably we should backport this all the way back to stable25.

File: src/tests/markdown.spec.js (hunk 1 above)
Comment (max-nextcloud) [reply src/tests/markdown.spec.js:8]: Good catch. Linter is not complaining though... funny.

File: src/components/PlainTextReader.vue [hunk 2]
```diff
@@ -37,7 +39,7 @@ export default {
 		renderHtml(content) {
 			return '<pre>' + escapeHtml(content) + '</pre>'
 		},
-		extensions: () => [PlainText],
+		extensions: () => [PlainText, CodeBlock],
```
Comment (juliushaertl) [inline src/components/PlainTextReader.vue:42]: Is this change related or shouldn't this be a separate PR?

**Review by juliushaertl — COMMENTED** [review]
Seems good just a comment about if the code block change is related or rather something for a separate PR.

File: src/components/PlainTextReader.vue (hunk 2 above)
Comment (max-nextcloud) [reply src/components/PlainTextReader.vue:42]: Good point... I'll split!

**Review by juliushaertl — APPROVED** [review]

**Comment by mejo-** [comment]
/backport 69d58894a94909068e1915179042a02c398a2e98,92a3dcf9eddf616b92c203c1843500624faa9b93 to stable27

**Comment by mejo-** [comment]
/backport 69d58894a94909068e1915179042a02c398a2e98,92a3dcf9eddf616b92c203c1843500624faa9b93 to stable26

**Comment by mejo-** [comment]
/backport 92a3dcf9eddf616b92c203c1843500624faa9b93 to stable25

**Comment by mejo-** [comment]
Manual backport to stable25: https://github.com/nextcloud/text/pull/4326
